packages/loot-core/src/client/accounts/accountsSlice.ts

```diff
@@ -0,0 +1,320 @@
+import { createSlice, type PayloadAction } from '@reduxjs/toolkit';
+
+import { send } from '../../platform/client/fetch';
+import { type AccountEntity, type TransactionEntity } from '../../types/models';
+import { addNotification, getAccounts, getPayees } from '../actions';
+import * as constants from '../constants';
+import { createAppAsyncThunk } from '../redux';
+import { type AppDispatch } from '../store';
+
+const sliceName = 'accounts';
+
+const initialState: AccountState = {
+  failedAccounts: {},
+  accountsSyncing: [],
+};
+
+type SetAccountsSyncingPayload = {
+  ids: Array<AccountEntity['id']>;
+};
+
+type MarkAccountFailedPayload = {
+  id: AccountEntity['id'];
+  errorType: string;
+  errorCode: string;
+};
+
+type MarkAccountSuccessPayload = {
+  id: AccountEntity['id'];
+};
+
+type AccountState = {
+  failedAccounts: {
+    [key: AccountEntity['id']]: { type: string; code: string };
+  };
+  accountsSyncing: Array<AccountEntity['id']>;
+};
+
+const accountsSlice = createSlice({
+  name: sliceName,
+  initialState,
+  reducers: {
+    setAccountsSyncing(
+      state,
+      action: PayloadAction<SetAccountsSyncingPayload>,
+    ) {
+      state.accountsSyncing = action.payload.ids;
+    },
+    markAccountFailed(state, action: PayloadAction<MarkAccountFailedPayload>) {
+      state.failedAccounts[action.payload.id] = {
+        type: action.payload.errorType,
+        code: action.payload.errorCode,
+      };
+    },
+    markAccountSuccess(
+      state,
+      action: PayloadAction<MarkAccountSuccessPayload>,
+    ) {
+      delete state.failedAccounts[action.payload.id];
+    },
+  },
+});
+
+type UnlinkAccountPayload = {
+  id: AccountEntity['id'];
+};
+
+export const unlinkAccount = createAppAsyncThunk(
+  'accounts/unlinkAccount',
+  async ({ id }: UnlinkAccountPayload, { dispatch }) => {
+    const { markAccountSuccess } = accountsSlice.actions;
+
+    await send('account-unlink', { id });
+    dispatch(markAccountSuccess({ id }));
+    await dispatch(getAccounts());
```

Comment:
For my own understanding, this is now awaited because `getAccounts` is a thunk, right? And it not being awaited before was a mistake in the code?

Reply:
That's true. It wasn't causing any issues before because there's nothing happening after the getAccounts call but ideally we should await thunks